deploy-service/common/src/main/java/com/pinterest/deployservice/scm/GithubManager.java

```diff
@@ -85,15 +85,19 @@ private void setHeaders() throws Exception {
                 GitHub gitHubApp = new GitHubBuilder().withJwtToken(jwtToken).build();
                 GHAppInstallation appInstallation = gitHubApp.getApp().getInstallationByOrganization(this.githubAppOrganization);
                 GHAppInstallationToken appInstallationToken = appInstallation.createToken().create();    
-                this.token = appInstallationToken.getToken();
+
+                // always use the newly created github app token as the token will expire
+                this.headers.put("Authorization", String.format("Token %s", appInstallationToken.getToken()));
             } catch (Exception e) {
                 // e.printStackTrace();
                 LOG.error("Exception when getting Github token: ", e);
                 throw e;
             }
         }
-
-        this.headers.put("Authorization", String.format("Token %s", this.token));
```

Comment:
what's fixed here ?

what's the issue with original code?

Before:
--------
if (token empty):
  get token

set header

After:
------
if (token empty):
  get token
  set header
else
  set header

Reply:
i think this is just a minor fix putting headers once, the main fix is removing the extra getToken()

Reply:
now auth header won't be put if there is no token. which seems fine? :)

Reply:
with change, will it ever cache token?

Reply:
with change, will it ever cache token?

Reply:
We no longer cache token as the original problem is that the GitHub token was cached and expired

Reply:
so do we still need else case and token variable ?

Reply:
this.token is populated from json. Are we putting token in config yaml?

Reply:
The current config does not use personal token in config yaml but the code supports that, which is the reason for the else case.